### PR TITLE
build: use updated pipelines from service-template

### LINKED ISSUE #22: Update to GH actions based on keptn-service-template-go
Please update this repository to use GH actions/workflows based on https://github.com/keptn-sandbox/keptn-service-template-go

### PR DESCRIPTION
**Pull request by christian-kreuzberger-dtx**
## This PR

- updates the CI pipeline to use gh actions
- introduces a renovate.json
- introduces a .versionrc.json

Fixes #22

### CODE REVIEW

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+  # always execute docker build when something is pushed to master or release-* branches
+  push:
+    branches:
+      - 'master'
+      - 'release-*'
+  # in addition, execute for pull requests to those branches
+  pull_request:
+    branches:
+      - 'master'
+      - 'release-*'
```
Comment (mowies) [inline .github/workflows/CI.yml:12]: If you're gonna use `standard-version` for releases I don't think you should trigger builds on `release-*` but rather on maintenance branches like `[0-9]+.[1-9][0-9]*.x`

File: Dockerfile [hunk 2]
```diff
@@ -49,20 +61,11 @@ RUN wget https://github.com/argoproj/argo-rollouts/releases/download/v0.10.2/kub
...
 ENV GOTRACEBACK=all
 
-# KEEP THE FOLLOWING LINES COMMENTED OUT!!! (they will be included within the travis-ci build)
-#travis-uncomment ADD MANIFEST /
-#travis-uncomment COPY entrypoint.sh /
-#travis-uncomment ENTRYPOINT ["/entrypoint.sh"]
-
-CMD ["/argo-service"]
+# Run the web service on container startup.
+CMD ["/prometheus-service"]
```
Comment (mowies) [inline Dockerfile:71]: copy & paste error :)

File: .github/workflows/unit-tests.yml [hunk 3]
```diff
@@ -0,0 +1,15 @@
+name: Unit Tests
```
Comment (mowies) [inline .github/workflows/unit-tests.yml:1]: Why do you have this pipeline if you already have `.github/actions/unit-tests.yml` and the CI pipeline?

File: .github/workflows/reviewdog.yml [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+      - name: Install linters
+        run: '( mkdir linters && cd linters && go get golang.org/x/lint/golint )'
+      - uses: reviewdog/action-setup@v1.0.3
+        with:
+          reviewdog_version: latest
+      - name: Run reviewdog
+        env:
+          REVIEWDOG_GITHUB_API_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          reviewdog -reporter=github-pr-review
```
Comment (mowies) [inline .github/workflows/reviewdog.yml:24]: missing newline :)

File: .github/workflows/reviewdog.yml [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+on: [pull_request]
+jobs:
+  reviewdog:
+    name: reviewdog
+    runs-on: ubuntu-latest
+    steps:
+      - name: Set up Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: 1.17
```
Comment (mowies) [inline .github/workflows/reviewdog.yml:11]: this is the only occurrence of Go v1.17. Everywhere else, you're using 1.16

File: .github/workflows/CI.yml [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+    runs-on: ubuntu-20.04
+    outputs: # declare what this job outputs (so it can be re-used for other jobs)
+      # build config
+      # metadata
+      GIT_SHA: ${{ steps.extract_branch.outputs.GIT_SHA }}
+      BRANCH: ${{ steps.extract_branch.outputs.BRANCH }}
+      BRANCH_SLUG: ${{ steps.extract_branch.outputs.BRANCH_SLUG }}
+      VERSION: ${{ steps.get_version.outputs.VERSION }}
+      DATE: ${{ steps.get_datetime.outputs.DATE }}
+      TIME: ${{ steps.get_datetime.outputs.TIME }}
```
Comment (mowies) [inline .github/workflows/CI.yml:30]: From what I can tell, `DATE` and `TIME` are never used. Only the `DATETIME` variable is used, you could remove the outputs and also the variables in the `get_datetime` step

File: .github/workflows/CI.yml [hunk 7]
```diff
@@ -0,0 +1,148 @@
...
+          BRANCH: ${{ steps.extract_branch.outputs.BRANCH }}
+          BRANCH_SLUG: ${{ steps.extract_branch.outputs.BRANCH_SLUG }}
+        shell: bash
+        run: |
+          # determine version
+          GIT_LAST_TAG=${{ steps.get_previous_tag.outputs.tag }}
+          GIT_NEXT_TAG=${{ steps.get_next_semver_tag.outputs.patch }}
+          echo "GIT_LAST_TAG=${GIT_LAST_TAG}, GIT_NEXT_TAG=${GIT_NEXT_TAG}"
+
+          if [[ "$BRANCH" == "release-"* ]]; then
```
Comment (mowies) [inline .github/workflows/CI.yml:70]: If you're using `standard-version` for releases, this condition will never be true and can probably be removed or replaced with the condition for maintenance branches
[Here](https://github.com/keptn/keptn/blob/master/.github/workflows/CI.yml#L191-L197) is the reference on how I adapted it with the new release-automation

**Review by mowies — CHANGES_REQUESTED** [review]

File: .github/workflows/CI.yml (hunk 1 above)
Comment (christian-kreuzberger-dtx) [reply .github/workflows/CI.yml:12]: Thanks. For now we will keep building from `release-*` branches.

File: .github/workflows/CI.yml (hunk 7 above)
Comment (christian-kreuzberger-dtx) [reply .github/workflows/CI.yml:68]: Thanks. For now we will keep building from `release-*` branches.

File: Dockerfile (hunk 2 above)
Comment (christian-kreuzberger-dtx) [reply Dockerfile]: fixed

File: .github/workflows/unit-tests.yml (hunk 3 above)
Comment (christian-kreuzberger-dtx) [reply .github/workflows/unit-tests.yml]: thanks, file removed

File: .github/workflows/reviewdog.yml (hunk 4 above)
Comment (christian-kreuzberger-dtx) [reply .github/workflows/reviewdog.yml]: fixed

File: .github/workflows/reviewdog.yml (hunk 5 above)
Comment (christian-kreuzberger-dtx) [reply .github/workflows/reviewdog.yml]: fixed

File: .github/workflows/CI.yml (hunk 6 above)
Comment (christian-kreuzberger-dtx) [reply .github/workflows/CI.yml]: good catch, removed.

**Review by mowies — APPROVED** [review]

**Pull request merged**